core/cap-0034.md

```diff
@@ -0,0 +1,352 @@
+constraints on a validator).  It would make more sense if they were only charged
+the fee that they were willing to pay that actually got them accepted into a
+transaction set.
+
+- Probably worse, there is a design pattern in some smart contracts which
+breaks, allowing inconsistent transactions to be committed, if it encounters
+this race:  a smart contract submits two transactions, `A` and `B`, which it
+intends to be mutually exclusive, and which it tries to ensure are mutually
+exclusive by marking `A` as being valid only up to some time `t` (it will fail
+with `txTOO_LATE` if it has not been committed by time `t`) and `B` as being
+valid only after time `t` (it will fail with `txTOO_EARLY` if it is accepted
+into a transaction set before `t`).  If the race that this CAP is intended to
+prevent occurs, as it can in the current protocol, then `A` might consume a
+sequence number but then fail.  But its consumption of the sequence number will
+allow transactions dependent on `A` (which the smart contract chains after `A`)
+to be executed, and if any of them succeeds, then they will perform operations
+which the smart contract intended to be performed only if the operations of `A`
+had been performed.  This race is probably uncommon, because smart contracts
+tend to set expiration times much farther out than SDKs (on the order of
+anywhere from hours to years), but potentially arbitrarily bad for a smart
+contract that did hit it.
+
+Additionally, externalizing transaction sets that contain significant numbers
+of transactions that are doomed to fail with `txTOO_LATE` during ledger close
+might mean that we're choosing sets that we would have considered worse than
+other available candidates if we had taken into account that those transactions
+were doomed.
+
+### Goals Alignment
+
+This CAP is aligned with the following Stellar Network Goals:
+
+- The Stellar Network should be secure and reliable.  Though the
+events that could lead to a smart contract executing inconsistently because of
+the problem that this CAP intends to address might be very rare, they might
+also be of a kind of which even a single event could significantly damage trust.
+A "reliable" network must try to avoid _any_ behavior which could appear
+sufficiently unpredictable to a client that they would be unlikely to consider
+it and could obtain inconsistent results when it happened.
+
+- The Stellar Network should run at scale and at low cost to all users.
+Keeping transactions which are guaranteed to fail later in
+externalized transaction sets is wasteful.
+
+It also aligns with the following Stellar Network Value:
+
+- The protocol should bias toward simplicity.  This CAP simplifies the
+ballot protocol by removing the "combination" of `closeTime`s and simply
+selecting the one associated with the selected transaction set.
+
+## Abstract
+
+We change the (deterministic) mechanism by which the ballot protocol computes
```

Comment:
It might be better if you broke this down into some simpler sentences, eg. "The protocol currently produces a candidate value for the ballot protocol by .... We propose producing a candidate value for the ballot protocol by ..." Just an example, but I think it would make this a lot easier to read.

Reply:
Yeah, wow, that sentence did get rather long.  I broke it up.